pyriemann/classification.py

```diff
@@ -498,19 +498,136 @@ def fit(self, X, y):
+        pred : ndarray of int, shape (n_matrices, 1)
+            Predictions for each matrix according to the closest centroid.
         """
         dist = self._predict_distances(covtest)
         neighbors_classes = self.classes_[np.argsort(dist)]
         out, _ = stats.mode(neighbors_classes[:, 0:self.n_neighbors], axis=1)
         return out.ravel()
+
+
+class SVC(BaseEstimator, ClassifierMixin):
+    """Classification by Riemannian Support Vector Machine.
+
+    Support vector machine with precomputed Riemannian kernel matrix
+    according to different metrics as described in [1]_.
+
+    Parameters
+    ----------
+    metric : {'riemann', 'euclid', 'logeuclid'}
+        Metric for kernel matrix computation.
+    Cref : None | ndarray, shape (n_channels, n_channels)
+        Reference point for kernel matrix computation. If None, the mean of
+        the training data according to the metric is used.
+    **kwargs
+        Keyword arguments passed to svc_func.
+
+    Attributes
+    ----------
+    svc_ : svc_func instance
+        Fitted SVC with precomputed Riemannian kernel matrix.
+    data_ : ndarray, shape (n_matrices, n_channels, n_channels)
+        If fitted, training data.
+
+    Notes
+    -----
+    .. versionadded:: 0.2.8
+
+    References
+    ----------
+    .. [1] A. Barachant, S. Bonnet, M. Congedo, and C. Jutten.
+        Classification of covariance matrices using a Riemannian-based kernel
+        for BCI applications". In: Neurocomputing 112 (July 2013), pp. 172-178.
+    """
+
+    def __init__(self, metric='riemann', Cref=None, **kwargs):
+        """Init."""
+        self.svc_func = sklearnSVC
+        self.Cref = Cref
+        self.metric = metric
+        self.svc_params = kwargs
+        self.svc_ = self.svc_func(kernel='precomputed', **self.svc_params)
+
+    def __setattr__(self, name, value):
+        """Enable setting attributes for SVC subclass."""
+        if 'svc_' in self.__dict__.keys():
+            if name in self.svc_.__dict__.keys():
+                self.svc_.set_params(**{name: value})
+                return
+        super().__setattr__(name, value)
+
+    def fit(self, X, y):
+        """Fit.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+        y : ndarray, shape (n_matrices, 1)
+            labels corresponding to each matrix.
+
+        Returns
+        -------
+        self : Riemannian SVC instance
+            The SVC instance.
+        """
+        kernelmat = kernel(X, Cref=self.Cref, metric=self.metric)
+        self.data_ = X
+        self.svc_ = self.svc_.fit(kernelmat, y)
+
+        return self
+
+    def predict(self, X):
+        """Get the predictions.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+
+        Returns
+        -------
+        pred : ndarray of int, shape (n_matrices, 1)
```

Comment:
```suggestion
        pred : ndarray of int, shape (n_matrices,)
```
it's what it should be